books/views.py

```diff
@@ -1,3 +1,89 @@
-from django.shortcuts import render
+import json
+from datetime               import datetime
+from json.decoder           import JSONDecodeError
 
-# Create your views here.
+from django.views           import View
+from django.http            import JsonResponse
+from django.db.models       import Q
+
+from books.models           import Book, BookStatus
+from places.models          import Place
+from users.utils            import login_required
+
+class BookPlaceView(View):
+    @login_required
+    def post(self, request):
+        try:
+            user       = request.user
+            data       = json.loads(request.body)
+            place_id   = data['place_id']
+            
+            if not Place.objects.filter(id=place_id).exists():
+                return JsonResponse({ "message":"INVALID_PLACE"}, status=404)
+            
+            start_time = datetime.strptime(data['start_time'],"%Y-%m-%dT%H:%M:%S")
+            end_time   = datetime.strptime(data['end_time'],"%Y-%m-%dT%H:%M:%S")
+            date       = start_time.date()
+
+            if end_time <= start_time:
+                return JsonResponse({"message":"INVALID_BOOK_TIME"}, status=400)
+
+            q1 = Q(start_time__gte=start_time) & Q(start_time__lt=end_time)
+            q2 = Q(end_time__gt=start_time) & Q(end_time__lte=end_time)
+            q3 = Q(start_time=start_time) & Q(end_time=end_time)           
+
+            if Book.objects.filter(place_id=place_id).filter(q1|q2|q3):
+                return JsonResponse({"message":"ALREADY_BOOKED_TIME"}, status=400)
```

Comment:
filter가 아니라 `exists()` method를 사용해서 구현해 주세요~
```suggestion
            if Book.objects.filter(place_id=place_id).filter(q1|q2|q3).exists():
                return JsonResponse({"message":"ALREADY_BOOKED_TIME"}, status=400)
```

Reply:
네 수정하겠습니다